camunda.Dockerfile

```diff
@@ -0,0 +1,166 @@
+# distribution than the official JRE. This will also include useful debugging tools like
+# jcmd, jmod, jps, etc., which take little to no space. Anecdotally, compressing modules add around
+# 10ms to the start up time, which is negligible considering our application takes ~10s to start up.
+# See https://adoptium.net/blog/2021/10/jlink-to-produce-own-runtime/
+# hadolint ignore=DL3018
+# required to compile a JRE on ARM64
+# see https://github.com/openzipkin/docker-java/issues/34
+ENV JAVA_TOOL_OPTIONS "-Djdk.lang.Process.launchMechanism=vfork"
+RUN jlink \
+     --add-modules ALL-MODULE-PATH \
+     --strip-debug \
+     --no-man-pages \
+     --no-header-files \
+     --compress=2 \
+     --output /jre && \
+   rm -rf /jre/lib/src.zip
+
+### Java base image
+FROM base AS java
+WORKDIR /
+
+# Inherit from previous build stage
+ARG JAVA_HOME=/opt/java/openjdk
+
+# Default to UTF-8 file encoding
+ENV LANG='C.UTF-8' LC_ALL='C.UTF-8'
+
+# Setup JAVA_HOME and binaries in the path
+ENV JAVA_HOME ${JAVA_HOME}
+ENV PATH $JAVA_HOME/bin:$PATH
+
+# Copy JRE from previous build stage
+COPY --from=jre-build /jre ${JAVA_HOME}
+
+# https://github.com/docker-library/openjdk/issues/212#issuecomment-420979840
+# https://openjdk.java.net/jeps/341
+# TL;DR generate some class data sharing for faster load time
+RUN java -Xshare:dump;
+
+### Build camunda from scratch ###
+FROM java as build
+WORKDIR /camunda
+ENV MAVEN_OPTS -XX:MaxRAMPercentage=80
+COPY --link . ./
+RUN --mount=type=cache,target=/root/.m2,rw \
+    ./mvnw -B -am -pl dist package -T1C -D skipChecks -D skipTests && \
+    mv dist/target/camunda-zeebe .
+
+### Extract camunda from distball ###
+# hadolint ignore=DL3006
+FROM base as distball
+WORKDIR /camunda
+ARG DISTBALL="dist/target/camunda-zeebe-*.tar.gz"
+COPY --link ${DISTBALL} camunda.tar.gz
+
+# Remove zbctl from the distribution to reduce CVE related maintenance effort w.r.t to containers
+RUN mkdir camunda-zeebe && \
+    tar xfvz camunda.tar.gz --strip 1 -C camunda-zeebe && \
+    find . -type f -name 'zbctl*' -delete
+
+### Image containing the camunda distribution ###
+# hadolint ignore=DL3006
+FROM ${DIST} as dist
+
+### Application Image ###
+# https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+# hadolint ignore=DL3006
+FROM java as app
+# leave unset to use the default value at the top of the file
+ARG BASE_IMAGE
+ARG BASE_DIGEST
+ARG VERSION=""
+ARG DATE=""
+ARG REVISION=""
+
+# OCI labels: https://github.com/opencontainers/image-spec/blob/main/annotations.md
+LABEL org.opencontainers.image.base.digest="${BASE_DIGEST}"
+LABEL org.opencontainers.image.base.name="docker.io/library/${BASE_IMAGE}"
+LABEL org.opencontainers.image.created="${DATE}"
+LABEL org.opencontainers.image.authors="zeebe@camunda.com"
+LABEL org.opencontainers.image.url="https://zeebe.io"
+LABEL org.opencontainers.image.documentation="https://docs.camunda.io/docs/self-managed/zeebe-deployment/"
```

Comment:
Probably should make those more generic

Reply:
for this one, we need a dedicate page in camunda-docs for the single app